### PR TITLE
CATROID-1123 Refactor ProjectUploadActivity and Controller to Kotlin

### PR DESCRIPTION
**Pull request by khushal1707**
Ticket: https://jira.catrob.at/browse/CATROID-1123

Refactored files:
- ProjectUploadActivity
- ProjectUploadController

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by Walcho1125 — APPROVED** [review]
Looks good to me!

File: catroid/src/main/java/org/catrobat/catroid/ui/controller/ProjectUploadController.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        val intent = Intent(context, ProjectUploadService::class.java)
+        val sceneNames = project.sceneNames.toTypedArray()
+        val resultReceiverWrapper = projectUploadInterface.getResultReceiverWrapper()
+        intent.putExtra(Constants.EXTRA_RESULT_RECEIVER, resultReceiverWrapper)
+        intent.putExtra(Constants.EXTRA_UPLOAD_NAME, projectName)
+        intent.putExtra(Constants.EXTRA_PROJECT_DESCRIPTION, projectDescription)
+        intent.putExtra(Constants.EXTRA_PROJECT_PATH, project.directory.absolutePath)
+        intent.putExtra(Constants.EXTRA_PROVIDER, Constants.NO_OAUTH_PROVIDER)
+        intent.putExtra(Constants.EXTRA_SCENE_NAMES, sceneNames)
+        return intent
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/controller/ProjectUploadController.kt:55]:
```suggestion
        return intent.apply {
           putExtra(Constants.EXTRA_RESULT_RECEIVER, resultReceiverWrapper)
           ...
        }
```

**Review by gPathpp — CHANGES_REQUESTED** [review]

File: catroid/src/main/java/org/catrobat/catroid/ui/ProjectUploadActivity.kt [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+            .setView(dialogUploadUnchangedProjectBinding.root)
+            .setPositiveButton(getString(R.string.ok)) { _: DialogInterface?, _: Int -> finish() }
+            .setCancelable(false)
+            .create()
+
+        alertDialog.show()
+    }
+
+    override fun onDestroy() {
+        if (uploadProgressDialog?.isShowing == true) {
```
[the remark this answers is not in the document]
Comment (gPathpp) [reply catroid/src/main/java/org/catrobat/catroid/ui/ProjectUploadActivity.kt:218]: Sorry, I didn't see that this Boolean can be null. So it's fine :+1:

**Review by gPathpp — APPROVED** [review]
LGTM

**Pull request merged**